cmd/dagger/cmd/common/common.go

```diff
@@ -82,6 +84,23 @@ func ValueDocOneLine(val *compiler.Value) string {
 	return strings.Join(docs, " ")
 }
 
+func convertRelativePaths(cacheOptionEntries []buildkit.CacheOptionsEntry) []buildkit.CacheOptionsEntry {
+    var convertedOptions = cacheOptionEntries
+    var pathableAttributes []string = []string{ "src", "dest" }
+
+    dirname, _ := os.UserHomeDir()
+
+    for _, option := range convertedOptions {
+        for _, key := range pathableAttributes {
+            if _, ok := option.Attrs[key]; ok {
+                option.Attrs[key] = strings.Replace(option.Attrs[key], "~", dirname, -1)
```

Comment:
We use https://github.com/mitchellh/go-homedir for this. You can look for examples by searching for `homedir.Expand`.